shimmy/dm_control_wrapper.py

```diff
@@ -0,0 +1,189 @@
+            spec[k] = dmc_spec2gym_space(v)
+        return spaces.Dict(spec)
+    elif isinstance(spec, dm_env.specs.BoundedArray):
+        low = np.broadcast_to(spec.minimum, spec.shape)
+        high = np.broadcast_to(spec.maximum, spec.shape)
+        return spaces.Box(low=low, high=high, shape=spec.shape, dtype=spec.dtype)
+    elif isinstance(spec, dm_env.specs.Array):
+        if np.issubdtype(spec.dtype, np.integer):
+            low = np.iinfo(spec.dtype).min
+            high = np.iinfo(spec.dtype).max
+        elif np.issubdtype(spec.dtype, np.inexact):
+            low = float("-inf")
+            high = float("inf")
+        else:
+            raise ValueError()
+
+        return spaces.Box(low=low, high=high, shape=spec.shape, dtype=spec.dtype)
+    else:
+        raise NotImplementedError
+
+
+def dmc_obs2gym_obs(obs):
+    """Converts a dm_control observation to a numpy array."""
+    if isinstance(obs, OrderedDict) or isinstance(obs, dict):
+        obs = copy.copy(obs)
+        for k, v in obs.items():
+            obs[k] = dmc_obs2gym_obs(v)
+        return obs
+    else:
+        return np.asarray(obs)
+
+
+class dm_control_wrapper(gym.Env):
+    """Wrapper that converts a dm_control environment into a gymnasium environment.
+    """
+
+
+    metadata = {"render_modes": ["human", "rgb_array"]}
+
+    def __init__(
+        self,
+        env: dm_env.Environment,
+        render_mode: str,
+        render_height: int = 84,
+        render_width: int = 84,
+        camera_id: int = 0,
+    ):
+        """Wrapper that converts a dm_control environment into a gymnasium environment.
+
+        Args:
+            env (dm_env.Environment): the base environment
+            render_mode (str): render mode of the environment, must be specified at start
+            render_height (int): the height of the render window
+            render_width (int): the width of the render window
+            camera_id (int): the ID of the camera that is used for rendering
+        """
+        self._env = env
+
+        # convert spaces
+        self.observation_space = dmc_spec2gym_space(self._env.observation_spec())
+        self.action_space = dmc_spec2gym_space(self._env.action_spec())
+
+        # camera rendering properties
+        self.render_mode = render_mode
+        self.render_height = render_height
+        self.render_width = render_width
+        self.camera_id = camera_id
+
+        # render viewer
+        if self.render_mode == "human":
+            from gymnasium.envs.mujoco.mujoco_rendering import Viewer
+
+            self.viewer = Viewer(
+                self._env.physics.model.ptr, self._env.physics.data.ptr
+            )
+
+    def __getattr__(self, name):
+        """__getattr__.
+
+        Args:
+            name:
+        """
+        return getattr(self._env, name)
+
+    def __repr__(self):
+        """__repr__.
+        """
+        description = f"All I can tell you is {self._env._task}."
```

Comment:
In gymnasium, environments only have `__str__` defined with `__repr__` calling `__str__`.
Could you change to be similar to the gymnasium environment styles, i.e. `DmControlCompatibility({self._env._task})`